### ISSUE TITLE
Add a state to the Loading component

### ISSUE THREAD

**Issue by Pierre-Monier** [IBM/carbon-components-svelte#692]
Hi, 

I want to add a feature to the Loading component. I want it to show the state of a task (for example, downloading an image). We will have to pass to value in order to do that. The total amount of data to download and the current amount of data downloaded.

Should I add this feature directly on the Loading component ? Or create a specific component ?

**Comment by metonym** [comment]
Have you considered customizing the value for the [`InlineLoading` description prop](https://carbon-svelte.vercel.app/components/InlineLoading#with-description)?

**Comment by Pierre-Monier** [comment]
No, good idea, it can do the trick. Thanks !

**Comment by metonym** [comment]
@Pierre-Monier There's actually a new [ProgressBar component](https://carbon-svelte.vercel.app/components/ProgressBar) released in v0.37.0 that might fit your use case:

<img width="825" alt="Screen Shot 2021-06-26 at 4 54 04 PM" src="https://user-images.githubusercontent.com/10718366/123528785-22fbbf00-d69f-11eb-8409-72aea493db67.png">

## [UX example (animated progress bar)](https://carbon-svelte.vercel.app/framed/ProgressBar/ProgressBarUx)

**Source code**

```svelte
<script>
  import { onMount } from "svelte";
  import { ProgressBar } from "carbon-components-svelte";

  let max = 328;
  let value = 0;
  let timer = undefined;

  onMount(() => {
    timer = setTimeout(() => {
      const interval = setInterval(() => {
        const delta = Math.random() * 10;

        if (value + delta < max) {
          value += delta;
        } else {
          value = max;
          clearInterval(interval);
        }
      }, 20);
    }, 2000);

    return () => clearTimeout(timer);
  });

  $: helperText =
    value > 0 ? `${value.toFixed(0)}MB of ${max}MB` : "Preparing upload...";
  $: if (value === max) helperText = "Done";
</script>

<ProgressBar
  labelText="Upload status"
  value="{value}"
  max="{max}"
  helperText="{helperText}"
/>
```

**Comment by Sheepux** [comment]
In addition i've just quick-tested a small modification of the imageLoader so that i can get the progress of the image download
downside of going through xhr, is that the image source is now a blob.
Looks like xhr is the only way to get an image loading progress.
https://github.com/manga-download/haruneko/tree/master/src/frontend/sample-svelte/lib

`<script>
    import ImageLoader from "../lib/imageLoader.svelte";
    let sizeloaded: number;
    let sizetotal: number;
</script>
            Download : {((sizeloaded / sizetotal) *100).toFixed(0)} %
            <ImageLoader
                fadein
                bind:sizeloaded
                bind:sizetotal
                src="https://effigis.com/wp-content/uploads/2015/02/DigitalGlobe_WorldView2_50cm_8bit_Pansharpened_RGB_DRA_Rome_Italy_2009DEC10_8bits_sub_r_1.jpg"
            >
                <div slot="loading">
                    the image is loading
                    {(sizeloaded / 1024 / 1024).toPrecision(2)} MB /
                    {(sizetotal / 1024 / 1024).toPrecision(2)} MB
                </div>
                <div slot="error">An error occurred.</div>
            </ImageLoader>
`

![custom-imageLoader](https://user-images.githubusercontent.com/19878135/123555366-9f68cd80-d785-11eb-839d-69d0bf6f7b22.gif)

I'm not sure if i should PR this (or maybe with an loader choice if end user wants the original behavior, but meaning no progress report ) :-\
